src/FlashMessage/FlashManager.php

```diff
@@ -50,22 +50,46 @@ public function __construct(SessionInterface $session, array $types, array $cssC
     /**
      * Tells if class may handle $object for status class rendering.
      *
+     * @param object|string $object     FlashManager or Sonata flash message type
+     * @param string|null   $statusName Sonata flash message type
+     *
      * @return bool
      */
     public function handlesObject($object, ?string $statusName = null)
     {
-        return \is_string($object) && \array_key_exists($object, $this->cssClasses);
+        if (\is_string($object)) {
+            if (null === $statusName) {
+                $statusName = $object;
+            }
+            $object = $this;
+        }
+
+        if (!$object instanceof self) {
+            return false;
+        }
+
+        return \array_key_exists($statusName, $this->cssClasses);
     }
 
     /**
      * Returns the status CSS class for $object.
      *
+     * @param object|string $object     FlashManager or Sonata flash message type
+     * @param string|null   $statusName Sonata flash message type
+     * @param string        $default    Default status class if flash message type do not exist
+     *
      * @return string
      */
     public function getStatusClass($object, ?string $statusName = null, string $default = '')
     {
-        return \array_key_exists($object, $this->cssClasses)
-            ? $this->cssClasses[$object]
+        if (\is_string($object)) {
+            if (null === $statusName) {
+                $statusName = $object;
+            }
+        }
+
+        return \array_key_exists($statusName, $this->cssClasses)
+            ? $this->cssClasses[$statusName]
             : $default;
```

Comment:
```suggestion
        return $this->cssClasses[$statusName] ?? $default;
```

Reply:
Sorry, watching again I think you want to allow `null` to be a valid value.
Dismiss my suggestion in that case.

Reply:
This can be do for FlashManager but not for StatusClassRenderer. I will mark it as `NEXT_MAJOR` in next RP 'Improve FlashManager`.